### DISCUSSION TITLE
How can I upgrade a custom release notes with renovate PRs

### DISCUSSION THREAD

**Discussion by onigoetz** [renovatebot/renovate#15424] [Request Help] [answered]
Hi,

is there a way to obtain a list of updated dependencies when running `postUpgradeTasks` ?

in my context, I'm trying to update a changelog file and automatically add a note on what was changed, my options are currently : 
* put a generic message that renovate upgraded something
* try to parse my `pom.xml` / `package.json` before and after changes and make a diff (doesn't seem very efficient)

Is there a possibility to do something like :

```
{
  "postUpgradeTasks": {
    "commands": ["npx @my-company/update-changelog {{rawUpdates}}"],
    "fileFilters": ["changelog.xml"],
    "executionMode": "branch"
  }
}

```

where `{{rawUpdates}}` would contain a stringified JSON of the updates ?

I'm open to other ways to do it, this is just a first idea to get the conversation started

**Comment by viceice** [comment, marked as answer]
checkout https://docs.renovatebot.com/self-hosted-configuration/#allowpostupgradecommandtemplating

`postUpgradeTasks` are only available when self-hosting renovate.

**Reply by onigoetz** [reply]
This is awesome ! thank you !

One thing: there is no link to this information from the `postUpgradeTasks` option page : https://docs.renovatebot.com/configuration-options/#postupgradetasks

**Reply by viceice** [reply]
feel free to open a PR to improve the docs 😊

**Reply by onigoetz** [reply]
sure : https://github.com/renovatebot/renovate/pull/15439
